CriteoMoPubAdapter/Sources/CriteoMoPubAdapter/CRNativeAdRenderer.m

```diff
@@ -0,0 +1,140 @@
+//
+//  CRNativeAdRenderer.m
+//  CriteoMoPubAdapter
+//
+//  Copyright © 2018-2020 Criteo. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "CRNativeAdRenderer.h"
+#import "CRNativeAdAdapter.h"
+#import <CriteoPublisherSdk/CriteoPublisherSdk.h>
+
+@interface CRNativeAdRenderer () <MPNativeAdRendererImageHandlerDelegate>
+
+@property(nonatomic, strong) CRNativeAdAdapter *adapter;
+@property(nonatomic, strong) CRNativeAdView<MPNativeAdRendering> *adView;
+@property(nonatomic, strong) Class renderingViewClass;
+@property(nonatomic) BOOL adViewInViewHierarchy;
+@property(nonatomic) MPNativeAdRendererImageHandler *rendererImageHandler;
+
+@end
+
+@implementation CRNativeAdRenderer
+
+- (instancetype)initWithRendererSettings:(id<MPNativeAdRendererSettings>)rendererSettings {
+  if (self = [super init]) {
+    MPStaticNativeAdRendererSettings *settings =
+        (MPStaticNativeAdRendererSettings *)rendererSettings;
+    _renderingViewClass = settings.renderingViewClass;
+    _viewSizeHandler = [settings.viewSizeHandler copy];
+    _rendererImageHandler = [MPNativeAdRendererImageHandler new];
+    _rendererImageHandler.delegate = self;
+  }
+  return self;
+}
+
++ (MPNativeAdRendererConfiguration *)rendererConfigurationWithRendererSettings:
+    (id<MPNativeAdRendererSettings>)rendererSettings {
+  MPNativeAdRendererConfiguration *config = [[MPNativeAdRendererConfiguration alloc] init];
+  config.rendererClass = [self class];
+  config.rendererSettings = rendererSettings;
+  config.supportedCustomEvents = @[ @"CRNativeCustomEvent" ];
+  return config;
+}
+
+- (UIView *)retrieveViewWithAdapter:(CRNativeAdAdapter<MPNativeAdAdapter> *)adapter
+                              error:(NSError *__autoreleasing *)error {
```

Comment:
NIT: I don't think `__autoreleasing` is needed as it is implicitly done by compiler